### PR TITLE
Updating geospatial datatype to return ogc wkb

### PR DESCRIPTION
**Pull request by sreekanth-db**
## Description

Replace standard JTS `WKBWriter` with custom `JTSOGCWKBWriter` to provide OGC-compliant WKB encoding with full support for 3D and 4D geometry coordinates (Z, M, and ZM dimensions).

**Changes:**
- Updated `WKTConverter.toWKB()` to use `JTSOGCWKBWriter` for OGC-compliant WKB generation
- Updated `WKTConverter.toWKT()` to preserve dimensions when converting WKB back to WKT

## Testing
Manually tested the flow with real databricks workspace and unit tests covering the following:

- 2D and 3D geometries: POINT/LINESTRING/POLYGON with Z and M coordinates
- 4D geometries: POINT/LINESTRING/POLYGON with both Z and M (ZM) coordinates
- Empty geometries and edge cases (negative coordinates, large decimal values)
- WKB → WKT → WKB and  WKB → WKT → WKB round-trip tests to verify binary-level consistency

*Update*: Have added a comprehensive test suite of 140+ test cases in `wkb_test_cases.json`

### CODE REVIEW

File: src/test/java/com/databricks/jdbc/api/impl/converters/WKTConverterTest.java [hunk 1]
```diff
@@ -203,4 +203,390 @@ public void testConcurrency() throws Exception {
       future.get();
     }
   }
+
+  // ========== 3D and 4D Geometry Tests (Z, M, ZM coordinates) ==========
+
+  @Test
+  public void testToWKB_PointZ_3D() throws DatabricksValidationException {
+    String wkt = "POINT Z (1 2 3)";
+    byte[] wkb = WKTConverter.toWKB(wkt);
```
[the remark this answers is not in the document]
Comment (mkaravel) [reply src/test/java/com/databricks/jdbc/api/impl/converters/WKTConverterTest.java]: I am also in favor of asserting the WKB values we get in all tests.

File: src/main/java/com/databricks/jdbc/api/impl/converters/WKTConverter.java [hunk 2]
```diff
@@ -22,15 +24,13 @@ public class WKTConverter {
...
-  private static final ThreadLocal<WKTWriter> WKT_WRITER = ThreadLocal.withInitial(WKTWriter::new);
   private static final ThreadLocal<WKBReader> WKB_READER = ThreadLocal.withInitial(WKBReader::new);
-  private static final ThreadLocal<WKBWriter> WKB_WRITER = ThreadLocal.withInitial(WKBWriter::new);
 
   /**
    * Converts WKT (Well-Known Text) to WKB (Well-Known Binary) format.
    *
    * <p>This implementation uses the JTS library to parse the WKT string into a Geometry object and
-   * then converts it to WKB format. This provides robust, standards-compliant conversion.
+   * then converts it to WKB format using the custom OGC-compliant WKB writer. geometry.
```
[the remark this answers is not in the document]
Comment (sreekanth-db) [reply src/main/java/com/databricks/jdbc/api/impl/converters/WKTConverter.java]: corrected the grammar

File: src/main/java/com/databricks/jdbc/api/impl/converters/WKTConverter.java [hunk 3]
```diff
@@ -43,7 +43,9 @@ public static byte[] toWKB(String wkt) throws DatabricksValidationException {
 
     try {
       Geometry geometry = WKT_READER.get().read(wkt);
-      return WKB_WRITER.get().write(geometry);
+      EnumSet<Ordinate> ordinates = determineOrdinates(geometry);
```
[the remark this answers is not in the document]
Comment (sreekanth-db) [reply src/main/java/com/databricks/jdbc/api/impl/converters/WKTConverter.java]: updated the implementation as per your suggestions

File: src/main/java/com/databricks/jdbc/api/impl/converters/WKTConverter.java [hunk 4]
```diff
@@ -68,14 +70,44 @@ public static String toWKT(byte[] wkb) throws DatabricksValidationException {
...
       LOGGER.error(errorMessage, e);
       throw new DatabricksValidationException(errorMessage, e);
     }
   }
 
+  private static EnumSet<Ordinate> determineOrdinates(Geometry geometry) {
+    // Handle empty geometries
+    if (geometry.isEmpty() || geometry.getCoordinate() == null) {
+      return EnumSet.of(Ordinate.X, Ordinate.Y);
+    }
```
[the remark this answers is not in the document]
Comment (sreekanth-db) [reply src/main/java/com/databricks/jdbc/api/impl/converters/WKTConverter.java]: updated the implementation as per your suggestions

File: src/test/java/com/databricks/jdbc/api/impl/converters/WKTConverterTest.java [hunk 5]
```diff
@@ -203,4 +204,431 @@ public void testConcurrency() throws Exception {
...
+    byte[] originalWkb = WKTConverter.toWKB(originalWkt);
+
+    String wkt = WKTConverter.toWKT(originalWkb);
+    byte[] finalWkb = WKTConverter.toWKB(wkt);
+
+    assertArrayEquals(originalWkb, finalWkb);
+  }
+
+  @Test
+  public void testRoundTrip_WKB_to_WKT_to_WKB_EmptyPoint() throws DatabricksValidationException {
```
[the remark this answers is not in the document]
Comment (sreekanth-db) [reply src/test/java/com/databricks/jdbc/api/impl/converters/WKTConverterTest.java]: Included these (or similar) tests cases in the wkb_test_cases.json
